### PR TITLE
feat: add an ApiResource PHP 8 attribute

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | no 
| New feature?  | yes 
| BC breaks?    | no
| Deprecations? | no
| Tickets       | n/a
| License       | MIT
| Doc PR        | todo

PHP 8 has been released today, let's celebrate!

```php
use ApiPlatform\Core\Annotation\ApiProperty;
use ApiPlatform\Core\Annotation\ApiResource;

#[ApiResource(description: "Hey PHP 8", iri: "https://php.net")]
class DummyPhp8
{
    #[ApiProperty(description: "We support attributes!")]
    public string $hello = 'hi';
}
```

### CODE REVIEW

File: src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php [hunk 1]
```diff
@@ -242,21 +243,19 @@ private function getPaginationDefaults(array $defaults, array $collectionPaginat
...
-            'graphql',
-        ];
+        $normalizedDefaults = ['attributes' => $defaults['attributes'] ?? []];
+        unset($defaults['attributes']);
+
+        [$publicProperties,] = ApiResource::getConfigMetadata();
 
         foreach ($defaults as $option => $value) {
-            if (\in_array($option, $rootLevelOptions, true)) {
+            if (isset($publicProperties[$option])) {
```
Comment (soyuka) [inline src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php:252]:
```suggestion
            if ($publicProperties[$option] ?? false) {
```

File: src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -598,7 +598,7 @@ private function addDefaultsSection(ArrayNodeDefinition $rootNode): void
                 return $normalizedDefaults;
             });
 
-        foreach (ApiResource::CONFIGURABLE_DEFAULTS as $attribute) {
+        foreach (ApiResource::getConfigMetadata()[1] as $attribute => $_) {
```
Comment (soyuka) [inline src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php:601]: [1] isn't so clear, I wish we could [, $configurableAttributes]; $configurableAttributes as ... :/

Comment (dunglas) [reply src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php:601]: We don't have something better for now.

File: src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php (hunk 1 above)
Comment (dunglas) [reply src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php:252]: it's longer :p

**Review by soyuka — APPROVED** [review]

**Pull request merged**